tests/file_storage/tests.py

```diff
@@ -729,6 +729,24 @@ def test_save_overwrite_behavior_temp_file(self):
         finally:
             self.storage.delete(name)
 
+    def test_file_name_truncation(self):
+        name = "test_long_file_name.txt"
+        file = ContentFile(b"content")
+        stored_name = self.storage.save(name, file, max_length=10)
+        try:
+            self.assertEqual(stored_name, "test_l.txt")
+            self.assertEqual(len(stored_name), 10)
+        finally:
+            self.storage.delete(stored_name)
```

Comment:
I know that we discussed this usage vs `addCleanup` but I see plenty of `addCleanup` calls, particularly in newer code. The usages of `try`-`finally` seem to be in older parts of the code:
```
$ grep finally tests.py | wc -l
7
$ grep addCleanup tests.py | wc -l
```
Would you agree to using the more modern pattern?